### ISSUE TITLE
use of NamedBlobFile breaks versioning

### ISSUE THREAD

**Issue by tkimnguyen** [plone/plone.app.versioningbehavior#38]
If you use NamedBlobFile fields in a content type, when you make changes to the content item and have versioning enabled you get this error stack:

```
2018-05-01 12:21:43 ERROR Zope.SiteErrorLog 1525195303.910.645064010487 http://localhost:8080/OIE/people/@@fc-paste
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module plone.app.content.browser.contents.paste, line 43, in __call__
  Module OFS.CopySupport, line 236, in manage_pasteObjects
  Module Products.BTreeFolder2.BTreeFolder2, line 461, in _setObject
  Module zope.event, line 31, in notify
  Module zope.component.event, line 24, in dispatch
  Module zope.component._api, line 136, in subscribers
  Module zope.component.registry, line 321, in subscribers
  Module zope.interface.adapter, line 585, in subscribers
  Module zope.component.event, line 32, in objectEventNotify
  Module zope.component._api, line 136, in subscribers
  Module zope.component.registry, line 321, in subscribers
  Module zope.interface.adapter, line 585, in subscribers
  Module plone.app.versioningbehavior.subscribers, line 62, in create_version_on_save
  Module Products.CMFEditions.CopyModifyMergeRepositoryTool, line 303, in save
  Module Products.CMFEditions.CopyModifyMergeRepositoryTool, line 448, in _recursiveSave
  Module Products.CMFEditions.ArchivistTool, line 265, in prepare
  Module Products.CMFEditions.ModifierRegistryTool, line 141, in getReferencedAttributes
  Module plone.app.versioningbehavior.modifiers, line 130, in getReferencedAttributes
AttributeError: 'NamedFile' object has no attribute 'open'
```

This is with Plone 5.0.8, plone.app.versioningbehavior 1.3, Products.CMFEditions 2.2.23, zope.interface 3.6.7, zope.component 3.9.5

```
    role_and_responsibility = field.NamedBlobFile(
        title=_('Role & Responsibility'),
        description=_(u'Upload a signed Program Liaison Role & Responsibilities form'),
    )
```

The workaround is to use NamedFile instead of NamedBlobFile.

**Comment by pysailor** [comment]
With this change plone/plone.app.contenttypes#480 that renamed the behaviour for `ILeadImage`, we also get this error on NewsItems in Plone sites that get migrated to 5.2.

I'm experimenting with a work-around that will try to fetch the actual blob from deep inside the shadowStorage. But that will mean that a lot of low-level functionality from CMFEditions will be used.

**Comment by ericof** [comment]
Closed with #45
